sample_cgfs/rv32i_priv.cgf

```diff
@@ -1,21 +1,413 @@
 ecall:
   config: 
     - check ISA:=regex(.*I.*); def rvtest_mtrap_routine=True 
-  opcode: 
+  mnemonics: 
     ecall: 0
 
 ebreak:
   config: 
     - check ISA:=regex(.*I.*); def rvtest_mtrap_routine=True 
-  opcode: 
+  mnemonics: 
     ebreak: 0
+    
+ 
+misa_check_combination:
+  config: 
+    - check ISA:=regex(.*I.*Zicsr.*F.*D); 
+  csr_comb:
+    (misa & 0x08) | (misa & 0x20) == 0 : 0			# setting “F”=0 and “D”=1 should results in both “F” and “D” being cleared.
+
+
+misa_mxl_check:
+  config: 
+    - check ISA:=regex(.*I.*Zicsr.*); 
+  csr_comb:
+    misa >> (xlen-2) == 0x01 : 0				# A coverpoint which checks whether the mxl field of misa register is 1.
+
+
+mstatus_Smode:    
+  config: 
+    - check ISA:=regex(.*I.*Zicsr.*); 
+  mnemonics: 
+    csrrw: 0
+    csrrs: 0
+  csr_comb:
+    (mstatus >> 1) & 0x01 == 0x01 : 0				# checking if the SIE gets enabled
+    (mstatus >> 5) & 0x01 == 0x01 : 0 			# checking SPIE bit 
+    (mstatus >> 8) & 0x01 == 0x01 : 0 			# checking SPP bit
+    (mstatus >> 22) & 0x1 == 0x01 : 0 			# When TSR=1, attempts to execute SRET while executing in S-mode will raise an illegal instruction exception.  
+    (mstatus >> 36) & 0x1 == 0x01 : 0 			# check SBE = 1, it controls whether explicit load and store memory accesses made from S-mode is big-endian  
```

Comment:
SBE can be hardwired to 0 (in most cases) which is a valid implementation. So this might have to move to a separate coverage label with a new config condition - big_endian_support

Reply:
But, isn't that true for a lot of CSR bits? For WARL fields we can't assume the bits are RW, so from a test perspective we should either be blindly trying to write them (Assuming that is it RW) and test that the right thing happens in either setting. If it isn't writable, then both Sail and DUT should be doing the same thing (eventually) and tests will pass.
Coverage will be missing - but as long as coverage of the RW version is covered during test development, we know that that some coverage be impossible when actually running tests.
The alternative is to condition tests depending on whether a CSR bit is RW or not and only select tests whose conditions can be met. That is similar to what you're proposing - but you can't stop at SBE.

Reply:
Edited